src/app/onion/dashboard/dashboard.component.ts

```diff
@@ -743,6 +750,38 @@ export class DashboardComponent implements OnInit, OnDestroy {
     });
   }
 
+  async openViewAllChangelogsModal(learningObjectId: string) {
+    this.openChangelogModal = true;
+    this.loadingChangelogs = true;
+    this.changelogLearningObject = this.learningObjects.find(learningObject => learningObject.id === learningObjectId);
+    try {
+      this.changelogs =  await this.changelogService.fetchAllChangelogs({
+        userId: this.changelogLearningObject.author.id,
+        learningObjectId: this.changelogLearningObject.id,
+      });
+    } catch (error) {
+      let errorMessage;
+
+      if (error.status >= 500) {
+        errorMessage = 'We encountered an error while attempting to retrieve changelogs for this learning object.';
+      } else if (error.status === 401) {
+        // user isn't logged-in, set client's state to logged-out and reload so that the route guards can redirect to login page
+        this.auth.logout();
+      } else {
+        errorMessage = 'An unknown error occurred while attempting to retrieve changelogs for this learning object.';
```

Comment:
What action should the user take because of this error?

Reply:
Resolved in 24ffadd